collector/manager/metrics_collector_manager.go

```diff
@@ -151,20 +151,33 @@ func (mcm *MetricsCollectorManager) declareMetricDefinitions(metricDetails []col
 
 	metricDefs := make([]hmetrics.MetricDefinition, len(endpoint.Metrics))
 
-	// For each metric in the endpoint, create a metric def for it.
-	// Notice metric tags override endpoint tags which override global tags.
-	// Do NOT allow pods to use agent environment variables since agent env vars may contain
-	// sensitive data (such as passwords). Only the global agent config can define tags
-	// with env var tokens.
-	globalTags := mcm.Config.Collector.Tags.ExpandTokens(true, additionalEnv)
-	endpointTags := endpoint.Tags.ExpandTokens(false, additionalEnv)
-
-	// we need these to expand tokens in the IDs
-	mappingFunc := expand.MappingFunc(false, additionalEnv)
-	mappingFuncWithEnv := expand.MappingFunc(true, additionalEnv)
-
 	for i, metric := range endpoint.Metrics {
 
+		// Define additional envvars with pod specific data for use in replacing ${env} tokens in tags.
+		env := map[string]string{
+			"METRIC:name":     	metric.Name,
+			"METRIC:id":	   	metric.ID,
+			"METRIC:units":	    	metric.Units,
+			"METRIC:description":	metric.Description,
+			"METRIC:type":		metric.Type,
```

Comment:
I am removing type (it was added right before I did the commit and is why its breaking the tests). Since the metric type is used to determine where the metric is going (eg counter/gauge) I don't think it makes sense to also store it as a tag for the metric itself.